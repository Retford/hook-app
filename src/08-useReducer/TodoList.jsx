import { TodoItem } from './TodoItem';

export const TodoList = ({ todos = [], onDeleteTodo, onToggleTodo }) => {
  return (
    <ul className='list-group'>
      {todos.map((todo) => (
        <div key={todo.id}>
          <TodoItem
            key={todo.id}
            todo={todo}
            onDeleteTodo={onDeleteTodo}
            onToggleTodo={onToggleTodo}
          />
        </div>
      ))}
    </ul>
  );
};
