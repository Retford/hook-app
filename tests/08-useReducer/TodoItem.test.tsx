import React from 'react';
import { render, screen, fireEvent, act } from '@testing-library/react';
import { TodoItem } from '../../src/08-useReducer/TodoItem';

describe('TodoItem', () => {
  const todo = [];
  const onDeleteTodo = jest.fn;
  const onToggleTodo = jest.fn;
  it('xd', async () => {
    const { container } = render(
      <TodoItem
        todo={todo}
        onDeleteTodo={onDeleteTodo}
        onToggleTodo={onToggleTodo}
      />
    );

    const deleteButton = await screen.findByRole('button');

    await act(async () => {
      fireEvent.click(deleteButton);
    });
    expect(container).toMatchSnapshot();

    screen.debug();
  });
});
