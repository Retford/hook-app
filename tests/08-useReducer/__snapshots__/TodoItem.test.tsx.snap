// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TodoItem xd 1`] = `
<div>
  <li
    class="list-group-item d-flex justify-content-between"
  >
    <span
      class="align-self-center  "
    />
    <button
      class="btn btn-danger"
    >
      Borrar
    </button>
  </li>
</div>
`;
